src/linker/Linker.Dataflow/MethodBodyScanner.cs

```diff
@@ -0,0 +1,825 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Collections.Generic;
+
+namespace Mono.Linker.Dataflow
+{
+	/// <summary>
+	/// Tracks information about the contents of a stack slot
+	/// </summary>
+	class StackSlot
```

Comment:
Kept this as `class` for now because that's what it is in .NET Native, but we should be able to make this a struct, I think.